### ISSUE TITLE
Rule improvement: `no-side-effects-in-computed-properties`

### ISSUE THREAD

**Issue by michalsnik** [vuejs/eslint-plugin-vue#97]
As @armano2 pointed in #58 we should consider supporting few more cases in this rule:

**1. Plugins provided by vuejs team, like:**

**Vue-router**
```js
export default {
  computed: {
    foo () {
      this.$router.go(-1)
      this.$router.push({})
      this.$router.replace({})
      return true
    }
  }
}
```
or **Vuex**
```js
export default {
  computed: {
    foo () {
      this.$store.commit({}) // Mutations
      this.$store.dispatch('', {}) // Actions
      return true
    }
  }
}
```

**2. Prototype methods**
```
return Array.prototype.unshift.apply(this.object, data) // i know this is invalid
return Object.assign(this.object, data);
```

There are probably more cases. We should gather them all here in this topic, discuss and choose which one are we going to support. Supporting different cases is good, but too many edge cases - not necessarily. We need to be smart about this.

**Comment by michalsnik** [comment]
Not a priority now. Current solution seems to be good enough.

**Comment by ylc395** [comment]
This issue should be re-opened.

**Comment by FloEdelmann** [comment]
Please open a new issue following the issue template and link to this issue from there, rather than commenting on an issue closed 7 years ago.
